lib/matplotlib/axis.py

```diff
@@ -2020,7 +2020,10 @@ def get_tick_space(self):
         # There is a heuristic here that the aspect ratio of tick text
         # is no more than 3:1
         size = tick.label1.get_size() * 3
-        return int(np.floor(length / size))
+        if size > 0:
+            return int(np.floor(length / size))
+        else:
```

Comment:
For what it is worth, keeping the else clause is about 10% faster. The order shouldn't matter and doesn't affect timings that I have noticed.

```console
scopatz@artemis ~ $ def f(x):
`·.,¸,.·*¯`·.,¸,.·*     if x:
`·.,¸,.·*¯`·.,¸,.·*         return 42
`·.,¸,.·*¯`·.,¸,.·*     else:
`·.,¸,.·*¯`·.,¸,.·*         return 0
`·.,¸,.·*¯`·.,¸,.·*     
`·.,¸,.·*¯`·.,¸,.·* 
scopatz@artemis ~ $ def g(x):
`·.,¸,.·*¯`·.,¸,.·*     if x:
`·.,¸,.·*¯`·.,¸,.·*         return 42
`·.,¸,.·*¯`·.,¸,.·*     return 0
`·.,¸,.·*¯`·.,¸,.·* 
scopatz@artemis ~ $ timeit! f(True)
10000000 loops, best of 3: 57.6 ns per loop
scopatz@artemis ~ $ timeit! g(True)
10000000 loops, best of 3: 63.6 ns per loop
scopatz@artemis ~ $ timeit! f(False)
10000000 loops, best of 3: 58 ns per loop
scopatz@artemis ~ $ timeit! g(False)
10000000 loops, best of 3: 64 ns per loop
scopatz@artemis ~ $ 1 - (58.0/64)
0.09375
```